### PR TITLE
Fix for REGISTRY-3840, REGISTRY-3843

### PR DESCRIPTION
**Pull request by madawas**
This PR is related to [REGISTRY-3840](https://wso2.org/jira/browse/REGISTRY-3840) and [REGISTRY-3843](https://wso2.org/jira/browse/REGISTRY-3843)

This PR enables the user to select multiple categorization options as well as improve categorization query building and also fixes the issue of categorization query replacing the existing query.

### CODE REVIEW

File: apps/store/themes/store/js/categorization.js [hunk 1]
```diff
@@ -82,43 +82,34 @@ var categorization = function () {
...
-                            } else {
-                                url = url + encodedQueryString;
-                            }
-                        } else {
-                            url = window.location.href + "%2C" + encodedQueryString;
-                        }
-                    } else {
-                        url = window.location.href + encodedQueryString;
-                    }
+                    debugger;
```
Comment (madawas) [inline apps/store/themes/store/js/categorization.js]: Removed unnecessary debugger entry in 04ebd24

File: apps/store/themes/store/js/categorization-data.js [hunk 2]
```diff
@@ -23,42 +23,62 @@
         if (typeof history.onpushstate == "function") {
             history.onpushstate({state: state});
         }
-        loadCategorizationEntries(url);
+        if (!state || state.categorization == false) {
```
Comment (splinter) [inline apps/store/themes/store/js/categorization-data.js]: You can change it to 
```js
if (!state && !state.categorization)
```

File: apps/store/themes/store/js/categorization-data.js [hunk 3]
```diff
@@ -23,42 +23,62 @@
...
-                        categorization();
-                        $('.refine > .panel > div').first().collapse('show');
-                    });
+/**
+ * This method loads the categorization-entry partial data when selection query changes. *
+ * @param url   url to load the data from
+ */
+function loadCategorizationEntries(url) {
+    caramel.data({
+        title: null,
```
Comment (splinter) [inline apps/store/themes/store/js/categorization-data.js]: Do we need to pass in title as null?

File: apps/store/themes/store/js/categorization-data.js (hunk 2 above)
Comment (madawas) [inline apps/store/themes/store/js/categorization-data.js]: @splinter Fixed in #841 (c7e7f2fba36a308ff02c339614352b54567c54fa)

File: apps/store/themes/store/js/categorization-data.js (hunk 3 above)
Comment (madawas) [inline apps/store/themes/store/js/categorization-data.js]: @splinter Thanks for pointing out. Removed this in #841 (c7e7f2fba36a308ff02c339614352b54567c54fa)

**Pull request merged**